nixos/modules/services/x11/xautolock.nix

```diff
@@ -131,6 +129,12 @@ in
           assertion = cfg.killer != null -> cfg.killtime >= 10;
           message = "killtime has to be at least 10 minutes according to `man xautolock`";
         }
-      ];
+      ] ++ (lib.flip map [ "locker" "notifier" "nowlocker" "killer" ]
+        (option:
+        {
+          assertion = cfg."${option}" != null -> builtins.substring 0 1 cfg."${option}" == "/";
```

Comment:
Can we take @rnhmjoj 's suggestion in #34371 ? i.e. use something like `if [[ -x ${cfg."${option}"} ]]`
`builtins.substring 0 1 cfg."${option}" == "/"` won't work if someone want `locker = "reboot"`.

Reply:
Yes, I think this should be done using a shell script,

Reply:
I agree that using "reboot" would not work just like that. Though one could easily specify an absolute path though:
```
services.xserver.xautlock.locker = "${pkgs.systemd}/bin/reboot";
````
Also this way it can be done using just nix itself and the path can be found by using `readlink $(which someCommand)` for example. I'm still open to suggestions :)

Reply:
Maybe there are some users already happily use xautolock service by setting `locker = "poweroff"`. If we only allow absolute path, this PR will force them to change their `configuration.nix`.
Also there is no downside if we uses `if [[ -x ${cfg."${option}"} ]]` to check if the value is an executable or not, right?
However, I think you are right and this is just a subtle improvement. The service will still work fine without the executable testing technique.

Reply:
@sifmelcara as far as I know this would not work just like that, as the `-x` flag checks if the following 'file' exists and is executable (see `man bash`). `"poweroff"` itself is neither a file nor an executable in that context and therefore the condition would evaluate to false:
```
if [[ -x poweroff ]]; then echo "it works"; else echo "it doesn't work"; fi
```
On the other hand, this would work if `poweroff` was replaced with `"$(which poweroff)"` in the condition.
I think that changing the module to expect a absolute paths wouldn't be much of a problem as the message in the assertion should give enough information on why the evaluation failed for someone who has something like `"poweroff"` in the config.

Reply:
Oh you are definitely right. Anyway, thank you for your great works!